addOns/automation/src/main/java/org/zaproxy/addon/automation/AuthenticationData.java

```diff
@@ -308,9 +335,20 @@ public void initContextAuthentication(
                                         ScriptBasedAuthenticationMethodType.SCRIPT_TYPE_AUTH,
                                         parameters.getOrDefault(PARAM_SCRIPT_ENGINE, "").toString(),
                                         progress);
-                        ScriptBasedAuthenticationMethodType scriptType =
-                                new ScriptBasedAuthenticationMethodType();
-                        ScriptBasedAuthenticationMethod scriptMethod =
+
+                        AuthenticationMethodType scriptType;
+
+                        if (getMethod()
+                                .toLowerCase(Locale.ROOT)
+                                .equals(AuthenticationData.METHOD_SCRIPT)) {
+                            scriptType = new ScriptBasedAuthenticationMethodType();
+                            LOGGER.debug("Loaded script auth method type");
+
+                        } else {
+                            scriptType = extAuth.getAuthenticationMethodTypeForIdentifier(8);
+                            LOGGER.info("Loaded client script auth method type {}.", scriptType);
```

Comment:
Why isn't this being done under `AuthenticationData.METHOD_CLIENT` instead?

Reply:
🤷‍♂️ because it's script 🤷‍♂️

Reply:
@psiinon I guess this might be why I was seeing things work the same with the weekly and without these changes?